build/kubernetes/k8s_setup.sh

```diff
@@ -11,9 +11,12 @@ curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/${KIND
 kind create cluster --image kindest/node:${K8S_VERSION}
 
 # Wait for cluster to be ready
-JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}';
-until kubectl get nodes -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do
-  sleep 1;
+JSONPATH='{range .items[*]}{@.metadata.name}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}'
+out=''
+until [[ "${out}" =~ 'Ready=True' ]]; do
+  sleep 1
+  out=$(kubectl get nodes -o jsonpath="$JSONPATH")
```

Comment:
I saw [here](https://github.com/kubernetes-sigs/kind/issues/4006#issue-3410403608) a better alternative for this logic?

```bash
echo "Waiting for cluster to be ready"
kubectl wait --for=condition=Ready nodes --all --timeout=60s >/dev/null 2>&1
```

AFAIK that is performing the same query? There script likewise calls `kind create cluster` just prior to this, but inlines a custom config:

```bash
echo "Creating Kind Cluster"
cat <<EOF | kind create cluster --config=- || exit 1
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  image: kindest/node:v${VERSION}
  extraPortMappings:
  - containerPort: 80
    hostPort: 80
    protocol: TCP
  - containerPort: 443
    hostPort: 443
    protocol: TCP
EOF
```

Reply:
FWIW I already [raised a PR for that change alone](https://github.com/coredns/ci/pull/168) given the age of this PR 😅 (_which doesn't inspire much hope for the PRs I raised for the CI to get merged_)